### PR TITLE
Implement IAsyncDisposable for DataContextTransaction

### PR DESCRIPTION
**Pull request by Shane32**
This adds an `IAsyncDisposable` implementation for `DataContextTransaction`s.

Note: `DataConnectionTransaction` already implements `IAsyncDisposable`

### CODE REVIEW

File: Source/LinqToDB/DataContextTransaction.cs [hunk 1]
```diff
@@ -11,6 +11,11 @@ namespace LinqToDB
 	/// </summary>
 	[PublicAPI]
 	public class DataContextTransaction : IDisposable
+#if NATIVE_ASYNC
+		, IAsyncDisposable
```
Comment (viceroypenguin) [inline Source/LinqToDB/DataContextTransaction.cs:15]: Any reason we can't add `using Async;` to the top of the file and avoid the ifdef here?

Comment (Shane32) [reply Source/LinqToDB/DataContextTransaction.cs:15]: The code here matches elsewhere in the library.  I can post another PR to change both places.

Comment (viceroypenguin) [reply Source/LinqToDB/DataContextTransaction.cs:15]: Sounds good. Thanks!

**Review by viceroypenguin — APPROVED** [review]

**Pull request merged**